src/steamship/agents/service/agent_service.py

```diff
@@ -338,6 +356,42 @@ def build_default_context(self, context_id: Optional[str] = None, **kwargs) -> A
         context = with_llm(context=context, llm=llm)
         return context
 
+    def _history_file_for_context(self, context_id: Optional[str] = None, **kwargs) -> File:
+        # AgentContexts serve to allow the AgentService to run agents
+        # with appropriate information about the desired tasking.
+        if context_id is None:
+            context_id = uuid.uuid4()
+
+        ctx = AgentContext.get_or_create(self.client, context_keys={"id": f"{context_id}"})
+        return ctx.chat_history.file
+
+    def _streaming_context_id_and_file(
+        self, context_id: Optional[str] = None, **kwargs
+    ) -> Tuple[Optional[str], File]:
+        history_file = self._history_file_for_context(context_id=context_id)
+        ctx_id = context_id
+
+        # if no context ID is provided, try to get a consistent ID from history_file
+        if not ctx_id:
+            ctx_id = _context_key_from_file(key="id", file=history_file)
+
+        # if you can't find a consistent context_id, then something has gone wrong, preventing streaming
+        if not ctx_id:
+            # TODO(dougreid): this points to a slight flaw in the context_keys vs. context_id
```

Comment:
Is this the situation where in practice we've just been using `context_id` whereas in theory we're hashing the whole `context_keys` object?

Reply:
yeah. we have a mismatch in our exposed args in `/prompt` vs. our actual capability. not sure if that matter practically yet.